.github/workflows/pr-checks.yml

```diff
@@ -0,0 +1,45 @@
+#
+# .github/workflows/pr-checks.yml
+#
+# https://docs.github.com/en/actions/reference/environment-variables#default-environment-variables
+---
+name: pr-checks
+on:
+  pull_request:
+    branches: [main]
+  push:
+    branches: [main]
+# https://docs.github.com/en/actions/using-jobs/using-concurrency
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+jobs:
+  lint:
+    runs-on: macos-14
+    defaults:
+      run:
+        shell: bash --login -eo pipefail {0}
```

Comment:
In order to source `.bashrc` the `--login` option is needed and the `--noprofile` option should **not** be specified. GHA `run` step default:

```
shell: /bin/bash --noprofile --login -e -o pipefail {0}
```